warp/aggregator/aggregation_job.go

```diff
@@ -0,0 +1,135 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package aggregator
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/set"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type signatureAggregationJob struct {
+	// SignatureBackend is assumed to be thread-safe and may be used by multiple signature aggregation jobs concurrently
+	client   SignatureBackend
+	height   uint64
+	subnetID ids.ID
+
+	quorumNum       uint64 // Minimum threshold at which to bother returning the resulting signature
+	cancelQuorumNum uint64 // Threshold at which to cancel further signature fetching
+	quorumDen       uint64 // Denominator to use when checking if we've reached the threshold
+	state           validators.State
+	msg             *avalancheWarp.UnsignedMessage
+}
+
+type AggregateSignatureResult struct {
+	SignatureWeight uint64
+	TotalWeight     uint64
+	Message         *avalancheWarp.Message
+}
+
+func NewSignatureAggregationJob(
+	client SignatureBackend,
+	height uint64,
+	subnetID ids.ID,
+	quorumNum uint64,
+	cancelQuorumNum uint64,
```

Comment:
Could we pick some more descriptive names? suggestions along the lines of `minAcceptableQuorumNum`, `maxNeededQuorumNum`

Reply:
Updating to `minAcceptableQuorumNum` and `minValidQuorumNum` and updating the comments